### PR TITLE
[VP][RISCV] Add vp.reduce.fmaximum/fminimum and its RISC-V codegen

### PR DESCRIPTION
**Pull request by mshockwave**
`vp.reduce.fmaximum/fminimum` are the VP version of `vector.reduce.fmaximum/fminimum`.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/vp-reduce.ll [hunk 1]
```diff
@@ -0,0 +1,205 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rvv/vp-reduce.ll:1]: This is an overly generic test name. How are the other vp.reduce tests structured?

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 2]
```diff
@@ -9804,8 +9811,25 @@ SDValue RISCVTargetLowering::lowerVPREDUCE(SDValue Op,
...
-  return lowerReductionSeq(RVVOpcode, Op.getSimpleValueType(), Op.getOperand(0),
-                           Vec, Mask, VL, DL, DAG, Subtarget);
+  SDValue Res =
+      lowerReductionSeq(RVVOpcode, Op.getSimpleValueType(), Op.getOperand(0),
+                        Vec, Mask, VL, DL, DAG, Subtarget);
+  if ((Opc != ISD::VP_REDUCE_FMINIMUM && Opc != ISD::VP_REDUCE_FMAXIMUM) ||
+      Op->getFlags().hasNoNaNs())
+    return Res;
+
+  // Propagate NaNs.
```
[the remark this answers is not in the document]
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:9822]: Good point, it's fixed now.

File: llvm/test/CodeGen/RISCV/rvv/vp-reduce.ll (hunk 1 above)
Comment (mshockwave) [reply llvm/test/CodeGen/RISCV/rvv/vp-reduce.ll]: I'd consolidates the tests with other floating point vp.reduce tests.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**